### PR TITLE
Add unit test coverage for `Rectangle::ClampUL`, and remove the unused `ClampLL` function

### PR DESCRIPTION
**Pull request by nlebeck**

### CODE REVIEW

**Comment by mitaclaw** [comment]
Regarding the "lower-right" origin rectangle put into your unit test, I'm not sure anything in the codebase relies upon an inside-out (left > right and/or bottom > top) rectangle being handled correctly. If this is true, it might be better to enforce the invariant that left <= right and top <= bottom for all `MathUtil::Rectangle`s, because then the `ClampUL` function can be optimized with this in mind.

Basically, I'm not sure if enshrining the existing behavior in unit tests is a good idea. It's good that this PR brought about this discussion, however.

**Comment by nlebeck** [comment]
Thanks for the review! I removed the final test case of a "lower-right" origin rectangle.

Agreed that it's better to enforce an invariant that a `MathUtil::Rectangle` is never inside-out if the codebase never wants to create an inside-out rectangle. I'd prefer to leave that change for a separate PR.

**Comment by mitaclaw** [comment]
@dolphin-emu-bot rebuild
